substrate/frame/revive/src/evm/tx_extension.rs

```diff
@@ -0,0 +1,100 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains transaction extensions needed for ethereum compatability.
+
+use crate::{CallOf, Config, Origin, OriginFor};
+use codec::{Decode, DecodeWithMemTracking, Encode};
+use frame_support::{
+	pallet_prelude::{InvalidTransaction, TransactionSource},
+	DebugNoBound, DefaultNoBound,
+};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	impl_tx_ext_default,
+	traits::{DispatchInfoOf, TransactionExtension, ValidateResult},
+	Weight,
+};
+
+/// An extension that sets the origin to [`Origin::EthTransaction`] in case it originated from an
+/// eth transaction.
+///
+/// This extension needs to be put behind any other extension that relies on a signed origin.
+#[derive(
+	Encode,
+	Decode,
+	DecodeWithMemTracking,
+	Clone,
+	Eq,
+	PartialEq,
+	DefaultNoBound,
+	TypeInfo,
+	DebugNoBound,
+)]
+#[scale_info(skip_type_params(T))]
+pub struct SetOrigin<T: Config + Send + Sync> {
+	/// Skipped as can only be set by runtime code.
+	#[codec(skip)]
```

Comment:
So this bit is the magic sauce. That's clever! The codec will always pick `false` as the default value here?